### DISCUSSION TITLE
How to determine the form factor of the device, whether it is a tablet or a phone?

### DISCUSSION THREAD

**Discussion by Pedr0Ross1** [AvaloniaUI/Avalonia#10724] [General]
I'm making apps with Avalonia and a know to determine if the Platform is Browser or Desktop, etc. But i can't determine if is tablet ou not. Googling i found that Xamarin or Uno have methods to do that. Maybe checking the screen size? I can't to find Screen Info or Display. I know classes like Screen or Screens, but i don't know how to use that with the current Screen or Primary Screen. Can anyone help me?

**Comment by workgroupengineering** [comment]
see [this](https://github.com/AvaloniaUI/Avalonia/blob/955a53595f7a4bcc1473f1d9e0a9a078cc146f3a/samples/ControlCatalog/Pages/PlatformInfoPage.xaml#L16)

**Reply by Pedr0Ross1** [reply]
Thaks for your reply. In IRuntimePlatform we can find some information about platform as well as others infos.
![image](https://user-images.githubusercontent.com/11988582/226690372-af304e62-7c20-4674-9fc1-6e7a9e13bf52.png)
All is just fine but i yet still retrive only of the device is Mobile, Desktop, Browser, Etc. How Can i determine if the device is a Mobile "Tablet" and what is the screen size. Position, orientation. Thanks.

**Reply by maxkatz6** [reply]
You shouldn't rely on screen size in this situation, and instead use your top level control or window size.

**Reply by Pedr0Ross1** [reply]
Thanks. Ok, but I know that must have a correct way to do this. I think so with the Screen class or alike. A can measure the window size or control,etc. Can you help me please. In the last case we need to discover the screen inch's or sizes. Or if you know a method to change app visual automatically for my controls to tablets or phones. I use different controls to phone or tablet, or general like desktop or browser.

**Reply by maxkatz6** [reply]
> I think so with the Screen class or alike

We have Screen API, but only for desktop for now. I don't think there is any way to get real screen size on mobile or browser.

> Or if you know a method to change app visual automatically for my controls to tablets or phones. I use different controls to phone or tablet, or general like desktop or browser.

Can you explain what **exactly** you want to change?
Normally, adaptive application detects input capabilities (touch, pen or mouse) and application size (not screen). Imagine browser web site, web page size can be smaller than screen size, so web content should adapt on the web page size instead.
To do so in Avalonia, you can subscribe on control.SizeChanged (or PropertyChanged and check for Bounds property, as SizeChanged was added only in latest previews for convenience). Input type can be detected by PointerMoved and similar events.

**Reply by Pedr0Ross1** [reply]
Thanks. I have made my own solution about the device information for platform specific to all platform in my own ImparSDK for Avalonia based on Uno Platform. ImparSDK Will soon available open source.

**Comment by Pedr0Ross1** [comment]
Thanks Everybody. I made my own solution about the device information for platform specific to all platform in my own ImparSDK for Avalonia based on Uno Platform. ImparSDK Will soon available open source.
